yt_dlp/extractor/afreecatv.py

```diff
@@ -380,3 +382,147 @@ def _real_extract(self, url):
+        broadcast_no = match.group('bno')
+
+        form_data = {
+            'bid': broadcaster_id,
+            'type': 'live',
+            'player_type': 'html5',
+            'stream_type': 'common',
+            'mode': 'landing',
+            'from_api': 0,
+        }
+        if broadcast_no:
+            form_data['bno'] = broadcast_no
+        info = self._download_json(
+            self._LIVE_API_URL,
+            broadcaster_id,
+            query={
+                'bjid': broadcaster_id,
+            },
+            data=urlencode_postdata(form_data),
+        )
+        channel_info = info['CHANNEL']
+        broadcast_no = channel_info.get('BNO') or broadcast_no
+        if not broadcast_no:
+            raise ExtractorError(
+                f'Unable to extract broadcast number ({broadcaster_id} may not be live)',
+                expected=True,
+            )
+
+        form_data['bno'] = broadcast_no
+        form_data['type'] = 'aid'
+        formats = []
+        quality_key = qualities(self._QUALITIES)
+        for quality_str in self._QUALITIES:
+            try:
+                form_data['quality'] = quality_str
+                aid_response = self._download_json(
+                    self._LIVE_API_URL,
+                    broadcaster_id,
+                    data=urlencode_postdata(form_data),
+                    note=f'Downloading access token for {quality_str} stream',
+                    errnote=f'Unable to download access token for {quality_str} stream',
+                )
+                aid = aid_response['CHANNEL']['AID']
+
+                stream_info_url = (
+                    channel_info.get('RMD', 'https://livestream-manager.afreecatv.com')
+                    + self._STREAM_API_PATH
+                )
+                cdn = channel_info.get('CDN', 'gcp_cdn')
+                stream_key = f'{broadcast_no}-common-{quality_str}-hls'
+                stream_info = self._download_json(
+                    stream_info_url,
+                    broadcast_no,
+                    query={
+                        'return_type': cdn,
+                        'use_cors': True,
+                        'cors_origin_url': 'play.afreecatv.com',
+                        'broad_key': stream_key,
+                    },
+                    note=f'Downloading metadata for {quality_str} stream',
+                    errnote=f'Unable to download metadata for {quality_str} stream',
+                    fatal=False,
+                )
+                view_url = stream_info['view_url']
+
+                formats.append({
+                    'format_id': quality_str,
+                    'quality': quality_key(quality_str),
+                    'ext': 'mp4',
+                    'protocol': 'm3u8',
+                    'url': update_url_query(view_url, {
+                        'aid': aid,
+                    }),
+                })
+            except (ExtractorError, KeyError):
+                continue
```

Comment:
remove try-except here aswell.

set `fatal=False` for api requests and then use `.get` or `try_get` to get the fields.
As of making sure shit is not appended to formats, make a check for all the necessary keys and only append if they are not None